commit 290f55520c62025da992d6be0002fc1e3738cfdb
Author: Noralf Trønnes <noralf@tronnes.org>
Date:   Sat Oct 3 15:58:59 2015 +0200

    dmaengine: bcm2835: set residue_granularity field
    
    bcm2835-dma supports residue reporting at burst level but didn't report
    this via the residue_granularity field.
    
    Without this field set properly we get playback issues with I2S cards.
    
    [by HiassofT, taken from bcm2708-dmaengine]
    Signed-off-by: Noralf Trønnes <noralf@tronnes.org>

diff --git a/drivers/dma/bcm2835-dma.c b/drivers/dma/bcm2835-dma.c
index b278c66..696fb30 100644
--- a/drivers/dma/bcm2835-dma.c
+++ b/drivers/dma/bcm2835-dma.c
@@ -802,6 +802,7 @@ static int bcm2835_dma_probe(struct platform_device *pdev)
 	od->ddev.src_addr_widths = BIT(DMA_SLAVE_BUSWIDTH_4_BYTES);
 	od->ddev.dst_addr_widths = BIT(DMA_SLAVE_BUSWIDTH_4_BYTES);
 	od->ddev.directions = BIT(DMA_DEV_TO_MEM) | BIT(DMA_MEM_TO_DEV);
+	od->ddev.residue_granularity = DMA_RESIDUE_GRANULARITY_BURST;
 	od->ddev.dev = &pdev->dev;
 	INIT_LIST_HEAD(&od->ddev.channels);
 	spin_lock_init(&od->lock);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1020_290f55520c62025da992d6be0002fc1e3738cfdb.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1020_290f55520c62025da992d6be0002fc1e3738cfdb.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
